stonesoup/detector/_video.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+import threading
+from abc import abstractmethod
+from copy import copy
+
+from .base import Detector
+from ..base import Property
+from ..buffered_generator import BufferedGenerator
+
+
+class _VideoAsyncDetector(Detector):
```

Comment:
Might want to name this `_VideoAsyncBoxDetector`. There exist algorithms that can generate masks and/or keypoints (see [here](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf1_detection_zoo.md) and [here](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md)).